grpc/src/client/channel.rs

```diff
@@ -228,6 +207,47 @@ impl PersistentChannel {
             runtime,
         }
     }
+
+    /// Returns the current state of the channel. If there is no underlying active channel,
+    /// returns Idle. If `connect` is true, will create a new active channel iff none exists.
+    fn state(
+        &self,
+        connect: bool,
+    ) -> Result<ConnectivityState, Box<dyn std::error::Error + Sync + Send>> {
+        let ac = self.get_active_channel(connect)?;
+        if let Some(s) = ac.connectivity_state.cur() {
+            return Ok(s);
+        }
+        return Ok(ConnectivityState::Idle);
+    }
+
+    /// Gets the underlying active channel. If `connect` is true, will create a new channel iff
+    /// there is no active channel.
+    fn get_active_channel(
+        &self,
+        connect: bool,
+    ) -> Result<Arc<ActiveChannel>, Box<dyn std::error::Error + Sync + Send>> {
+        let mut s = self
```

Comment:
Maybe name this `ac` or `active_channel`?  I'm not sure what `s` represents.

Reply:
Done!